cypress/integration/index.test.js

```diff
@@ -0,0 +1,55 @@
+describe("Index", () => {
+  beforeEach(() => {
+    cy.visit("./index.html");
+  });
+
+  it("loads ten jokes at first", () => {
+    cy.get(".card").its("length").should("eq", 10);
+  });
+
+  it("expands a card when clicked", () => {
+    let originalTitle;
+    cy.get(".card")
+      .first()
+      .then(($card) => {
+        originalTitle = $card.children(":visible").text();
+        return $card;
+      })
+      .click()
+      .should(($card) => {
+        expect($card.children(":visible").text()).not.to.be.equal(originalTitle);
+      });
+  });
+
+  it("contracts a card when clicked a second time", () => {
+    let originalTitle;
+    cy.get(".card")
+      .first()
+      .then(($card) => {
+        originalTitle = $card.children(":visible").text();
+        return $card;
+      })
+      .click()
+      .click()
+      .should(($card) => {
+        expect($card.children(":visible").text()).to.be.equal(originalTitle);
+      });
+  });
+
+  it("reloads new jokes when the reloader is clicked", () => {
+    let originalTitle;
+    cy.get(".card")
+      .first()
+      .then(($card) => {
+        originalTitle = $card.children(":visible").text();
+        return $card;
+      })
+      .get("#reload")
+      .click()
+      .get(".card")
+      .first()
+      .should(($card) => {
+        expect($card.children(":visible").text()).not.to.be.equal(originalTitle);
+      });
```

Comment:
Mmhh yeah good callout. I would normally want to use `cy.intercept` to switch out the network requests but the bug linked in the bonus makes that hard... Otherwise yes, we could do that!

For now I added a `retries: 2` to make this _really_ improbable. Think that's good enough?

Reply:
Yes, totally, great solution, thank you!